engine/src/main/java/com/agiletec/aps/system/common/tree/TreeNode.java

```diff
@@ -18,197 +18,223 @@
 	}
-	public void setChildren(ITreeNode[] children) {
-		this._children = children;
+
+	public void setChildrenCodes(String[] childrenCodes) {
+		this._childrenCodes = childrenCodes;
 	}
-	
+
 	/**
-	 * Adds a node to nodes in a lower level. 
-	 * The new node is inserted in the final position.
+	 * Adds a node to nodes in a lower level. The new node is inserted in the
+	 * final position.
+	 *
 	 * @param treeNode The node to add.
 	 */
-	public void addChild(ITreeNode treeNode) {
-		int len = this._children.length;
-		ITreeNode[] newChildren = new ITreeNode[len + 1];
-		for (int i=0; i < len; i++) {
-			newChildren[i] = this._children[i];
+	public void addChildCode(String treeNode) {
+		int len = this._childrenCodes.length;
+		String[] newChildren = new String[len + 1];
+		for (int i = 0; i < len; i++) {
+			newChildren[i] = this._childrenCodes[i];
 		}
 		newChildren[len] = treeNode;
-		this._children = newChildren;
+		this._childrenCodes = newChildren;
 	}
-	
+
 	@Override
 	public int getPosition() {
 		return _position;
 	}
+
 	protected void setPosition(int position) {
 		this._position = position;
 	}
-	
+
 	@Override
 	public ApsProperties getTitles() {
 		return _titles;
 	}
-	
+
 	/**
-	 * Set the titles of the node. 
-	 * @param titles A set of properties with the titles, 
-	 * where the keys are the codes of language.
+	 * Set the titles of the node.
+	 *
+	 * @param titles A set of properties with the titles, where the keys are the
+	 * codes of language.
 	 */
 	public void setTitles(ApsProperties titles) {
 		this._titles = titles;
 	}
-	
+
 	@Override
 	public void setTitle(String langCode, String title) {
 		this.getTitles().setProperty(langCode, title);
 	}
-	
+
 	@Override
 	public String getTitle(String langCode) {
 		return this.getTitles().getProperty(langCode);
 	}
-	
+
 	@Override
 	public String getFullTitle(String langCode) {
 		return this.getFullTitle(langCode, " / ");
 	}
-	
+
 	@Override
 	public String getFullTitle(String langCode, String separator) {
 		return this.getFullTitle(langCode, separator, false);
 	}
-	
+
 	@Override
 	public String getShortFullTitle(String langCode) {
 		return this.getShortFullTitle(langCode, " / ");
 	}
-	
+
 	@Override
 	public String getShortFullTitle(String langCode, String separator) {
 		return this.getFullTitle(langCode, separator, true);
 	}
-	
+
 	protected String getFullTitle(String langCode, String separator, boolean shortTitle) {
 		String title = this.getTitles().getProperty(langCode);
-		if (null == title) title = this.getCode();
-		if (this.isRoot()) return title;
+		if (null == title) {
+			title = this.getCode();
+		}
+		if (this.isRoot()) {
+			return title;
+		}
 		ITreeNode parent = this.getParent();
 		while (parent != null && parent.getParent() != null) {
 			String parentTitle = "..";
 			if (!shortTitle) {
 				parentTitle = parent.getTitles().getProperty(langCode);
-				if (null == parentTitle) parentTitle = parent.getCode();
+				if (null == parentTitle) {

```

Comment:
Can you add a short comment about the purpose of the whole while {} loop? Looks like the goal is to traverse the tree in reverse and build up a title but future devs may need a little bit of a guidance.

Reply:
Ok, I'll write a java comment